backend/src/main/java/com/bakdata/conquery/models/jobs/ImportJob.java

```diff
@@ -110,10 +86,25 @@ public static ImportJob create(
 			}
 
 			final ImportId importId = new ImportId(table.getId(), header.getName());
-
-			if (namespace.getStorage().getImport(importId) != null) {
-				throw new WebApplicationException(String.format("Import[%s] is already present.", importId), Response.Status.CONFLICT);
+			IdMutex.Locked locked = runningImportJobs.acquire(importId);
```

Comment:
So wie ich das sehe, brauchst du glaube ich gar kein Lock, denn alle Messages die du hier sendest sind Slow messages, heißt sie sind garantiert in-order verarbeitet zu werden.

Ich glaube das versenden des removeImportJobs könntest du als boolean-flag in die Prameter-Liste machen, dann vereinheitlichen sich auch deine  Branches.

Reply:
@awildturtok  der Lock ist eigentlich dafür da, auf die Beendigung des alten ImportJobs zu warten bevor der neue gestartet wird. 

wie meinst du das als Boolean-Flag?

Reply:
Genau, aber den Lock brauchst du eigentlich nicht, wenn alle Jobs nacheinander verarbeitet werden (was sie werden)